[Doc]Added Mooncake documentation

xiaguan opened the pull request:
Added Mooncake documentation

Tested locally and confirmed working

Future plans include submitting documentation for disaggregated prefill deployment

xiaguan committed:
docs: Add comprehensive MoonCake documentation

- Expanded mooncake.rst from placeholder to comprehensive guide
- Added architecture diagram (mooncake-store-preview.png)
- Included setup, configuration, and deployment instructions
- Added troubleshooting and additional resources sections

Signed-off-by: Jinyang Su <751080330@qq.com>

Shaoting-Feng reviewed (approved):
LGTM

maobaolong commented on docs/source/kv_cache/mooncake.rst:
```diff
@@ -1,11 +1,226 @@
+Additional Resources
+--------------------
+
+- `MoonCake Store Architecture <https://github.com/kvcache-ai/Mooncake/blob/main/doc/en/mooncake-store-preview.md>`_
```
As Mooncake is a word already, so it could be better to use `Mooncake` here

xiaguan replied:
Thks. Resolved

maobaolong commented on docs/source/kv_cache/mooncake.rst:
```diff
@@ -1,11 +1,226 @@
+    max_local_cpu_size: 5
+
+    extra_config:
+      local_hostname: "localhost"
```
Sorry for did not suggest here before, but #806.

It could be conflict to the extra_configs of other connectors, so, i suggest to add a prefix to each mooncakestore extra config, something like
```yaml
    extra_config:
      mooncakestore.local_hostname: "localhost"
```

And you don't need to address this in this PR.

xiaguan replied:
OK, I'll take care of it in a separate PR.

maobaolong reviewed (commented):
Thanks for this document, it will helps new users easy to use lmcache + mooncakestore.

Left minor comments.

xiaguan committed:
docs: update mooncake documentation for clearer branding

Signed-off-by: Jinyang Su <751080330@qq.com>

xiaguan committed:
docs: fix Mooncake capitalization consistency in documentation

- Standardize product name to 'Mooncake' (capital M, lowercase c)
- Update title, descriptions, and parameter documentation
- Maintain lowercase for technical components (mooncake_master, etc.)
- Ensure consistent naming throughout the documentation

Signed-off-by: Jinyang Su <751080330@qq.com>

maobaolong reviewed (approved):
@xiaguan It LGTM now, thanks for your contribution!

maobaolong committed:
Merge branch 'dev' into update_mooncake_doc
